// @my-scope/wryan21.multiflex-prime.ui.button/dist/button.composition.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisabledButton = exports.GhostIconOnlyButton = exports.GhostIconButton = exports.GhostButton = exports.IconOnlyButton = exports.IconButton = exports.TextButton = void 0;
const react_1 = __importDefault(require("react"));
const button_1 = require("./button");
const TextButton = () => (react_1.default.createElement(button_1.Button, { text: "act through me", icon: "", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.TextButton = TextButton;
const IconButton = () => (react_1.default.createElement(button_1.Button, { text: "act through me", icon: "creation", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.IconButton = IconButton;
const IconOnlyButton = () => (react_1.default.createElement(button_1.Button, { icon: "creation", text: "", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.IconOnlyButton = IconOnlyButton;
const GhostButton = () => (react_1.default.createElement(button_1.Button, { text: "act through me", variance: "ghost", icon: "", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.GhostButton = GhostButton;
const GhostIconButton = () => (react_1.default.createElement(button_1.Button, { text: "act through me", variance: "ghost", icon: "creation", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.GhostIconButton = GhostIconButton;
const GhostIconOnlyButton = () => (react_1.default.createElement(button_1.Button, { variance: "ghost", icon: "creation", text: "", disabled: false, disabledMsg: "", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.GhostIconOnlyButton = GhostIconOnlyButton;
const DisabledButton = () => (react_1.default.createElement(button_1.Button, { variance: "ghost", icon: "", text: "act through me", disabled: true, disabledMsg: "test message", tabIndex: 0, onclick: function () {
        throw new Error('Function not implemented.');
    } }));
exports.DisabledButton = DisabledButton;
